frame/staking-rewards/src/test/test_reward_accumulation_hook.rs

```diff
@@ -0,0 +1,418 @@
+		assert_last_event::<Test>(Event::StakingRewards(
+			crate::Event::<Test>::MaxRewardsAccumulated { pool_id: POOL_ID, asset_id: PICA::ID },
+		));
+	})
+}
+
+#[test]
+// takes about 3 minutes to run
+fn test_acumulate_rewards_hook() {
+	new_test_ext().execute_with(|| {
+		type A = Currency<97, 12>;
+		type B = Currency<98, 12>;
+		type C = Currency<99, 12>;
+		type D = Currency<100, 12>;
+		type E = Currency<101, 12>;
+		type F = Currency<102, 12>;
+
+		const fn block_seconds(block_number: u64) -> u128 {
+			((MILLISECS_PER_BLOCK / 1_000) * block_number) as u128
+		}
+
+		const ONE_YEAR_OF_BLOCKS: u64 = 60 * 60 * 24 * 365 / (block_seconds(1) as u64);
+
+		let mut current_block = System::block_number();
+
+		// 0.000_002 A per second, capped at 0.1 A
+		// cap will be hit after 50_000 seconds (8334 blocks)
+		let a_a_reward_rate = A::units(2) / 1_000_000;
+		let a_a_max_rewards = A::units(1) / 10;
+
+		// 0.000_002 B per second, capped at 0.05 B
+		// cap will be hit after 25_000 seconds (4167 blocks)
+		let a_b_reward_rate = B::units(2) / 1_000_000;
+		let a_b_max_rewards = B::units(5) / 100;
+
+		// 2 D per second, capped at 100k D
+		// cap will be hit after 5_000 seconds (834 blocks)
+		let c_d_reward_rate = D::units(2);
+		let c_d_max_rewards = D::units(10_000);
+
+		// 0.005 E per second, capped at 10 E
+		// cap will be hit after 2_000 seconds (334 blocks)
+		let c_e_reward_rate = E::units(5) / 1_000;
+		let c_e_max_rewards = E::units(10);
+
+		let cfgs = [
+			RewardPoolConfiguration::RewardRateBasedIncentive {
+				owner: ALICE,
+				asset_id: A::ID,
+				end_block: current_block + ONE_YEAR_OF_BLOCKS,
+				reward_configs: [
+					(
+						A::ID,
+						RewardConfig {
+							asset_id: A::ID,
+							max_rewards: a_a_max_rewards,
+							reward_rate: RewardRate::per_second(a_a_reward_rate),
+						},
+					),
+					(
+						B::ID,
+						RewardConfig {
+							asset_id: B::ID,
+							max_rewards: a_b_max_rewards,
+							reward_rate: RewardRate::per_second(a_b_reward_rate),
+						},
+					),
+				]
+				.into_iter()
+				.try_collect()
+				.unwrap(),
+				lock: default_lock_config(),
+			},
+			RewardPoolConfiguration::RewardRateBasedIncentive {
+				owner: BOB,
+				asset_id: C::ID,
+				end_block: current_block + ONE_YEAR_OF_BLOCKS,
+				reward_configs: [
+					(
+						D::ID,
+						RewardConfig {
+							asset_id: D::ID,
+							max_rewards: c_d_max_rewards,
+							reward_rate: RewardRate::per_second(c_d_reward_rate),
+						},
+					),
+					(
+						E::ID,
+						RewardConfig {
+							asset_id: E::ID,
+							max_rewards: c_e_max_rewards,
+							reward_rate: RewardRate::per_second(c_e_reward_rate),
+						},
+					),
+				]
+				.into_iter()
+				.try_collect()
+				.unwrap(),
+				lock: default_lock_config(),
+			},
+		];
+
+		for cfg in cfgs {
+			StakingRewards::create_reward_pool(Origin::root(), cfg).unwrap();
+		}
+
+		fn check_rewards(expected: &[(Public, u128, &[(u128, u128)])]) {
```

Comment:
Could you add a name comment to each anon field here? It would be easier to understand

Reply:
good idea, will add